internal/admission/admission_test.go

```diff
@@ -91,21 +95,21 @@ func TestCredentialName(t *testing.T) {
 			description:  "generated credentialName within character limit",
 			namespace:    "devops",
 			name:         "example-gateway",
-			wantContains: "devops-example-gateway-",
-			wantLen:      33,
+			wantContains: fmt.Sprintf("devops-example-gateway-%s", portName),
+			wantLen:      28,
 		},
 		{
 			description: "generated credentialName is truncated",
 			namespace:   strings.Repeat("a", 125),
 			name:        strings.Repeat("b", 125),
 			// some characters from the end of name should be truncated
-			wantContains: fmt.Sprintf("%s-%s-", strings.Repeat("a", 125), strings.Repeat("b", 116)),
+			wantContains: fmt.Sprintf("%s-%s-%s", strings.Repeat("a", 125), strings.Repeat("b", 121), portName),
 			wantLen:      secretNameMaxLength,
 		},
 	}
 
 	for _, test := range tests {
-		n := credentialName(context.TODO(), test.namespace, test.name)
+		n := credentialName(context.TODO(), test.namespace, test.name, portName)
 
 		assert.Contains(t, n, test.wantContains, test.description)
```

Comment:
Since this is now deterministic without the random chars, it can be changed to `assert.Equal`. `wantContains` can be renamed `want`